examples/functional_test.py

```diff
@@ -0,0 +1,279 @@
+                        duration,
+                        options.timeout,
+                    )
+                elif options.timeout:
+                    logger.info(
+                        "Stream completed naturally in %.1fs (max: %.1fs)",
+                        duration,
+                        options.timeout,
+                    )
+
+    return (sent_counter[0], received_count, error_count)
+
+
+async def main() -> int:
+    """Run the OAuth classification example."""
+    logger = logging.getLogger(__name__)
+    load_dotenv()
+
+    # Configuration
+    max_test_images = int(os.getenv("TEST_IMAGE_COUNT", str(10_000)))
+    min_interval_ms = os.getenv("TEST_MIN_INTERVAL_MS", None)
+    if min_interval_ms is not None:
+        min_interval_ms = int(min_interval_ms)
+
+    # OAuth credentials from environment
+    client_id = os.getenv("OAUTH_CLIENT_ID")
+    client_secret = os.getenv("OAUTH_CLIENT_SECRET")
+    auth_url = os.getenv(
+        "OAUTH_AUTH_URL", "https://crispthinking.auth0.com/oauth/token"
+    )
+    audience = os.getenv("OAUTH_AUDIENCE", "crisp-athena-dev")
+
+    if not client_id or not client_secret:
+        logger.error("OAUTH_CLIENT_ID and OAUTH_CLIENT_SECRET must be set")
+        return 1
+
+    host = os.getenv("ATHENA_HOST", "localhost")
+    logger.info("Connecting to %s", host)
+
+    # Create credential helper
+    credential_helper = CredentialHelper(
+        client_id=client_id,
+        client_secret=client_secret,
+        auth_url=auth_url,
+        audience=audience,
+    )
+
+    # Test token acquisition
+    try:
+        logger.info("Acquiring OAuth token...")
+        token = await credential_helper.get_token()
+        logger.info("Successfully acquired token (length: %d)", len(token))
+    except Exception:
+        logger.exception("Failed to acquire OAuth token")
+        return 1
+
+    max_deployment_id_length = 63
+
+    deployment_id = f"smoke-test-{uuid.uuid4()}"
+    if len(deployment_id) > max_deployment_id_length:
+        deployment_id = deployment_id[:max_deployment_id_length]
+
+    logger.info("Using deployment: %s", deployment_id)
+
+    # Run classification with OAuth authentication
+    options = AthenaOptions(
+        host=host,
+        resize_images=True,
+        deployment_id=deployment_id,
+        compress_images=True,
+        timeout=120.0,  # Maximum duration, not forced timeout
+        keepalive_interval=30.0,  # Longer intervals for persistent streams
+        affiliate="Crisp",
+    )
+
+    sent, received, errors = await run_smoke_test(
+        logger, options, credential_helper, max_test_images, min_interval_ms
+    )
+
+    if errors > 0:
+        logger.warning("Completed Unsuccessfully with %d errors", errors)
+        return 1
+
+    if sent == received:
+        logger.info("Success: %d requests processed", sent)
+        return 0
+
+    logger.warning("Incomplete: %d sent, %d received", sent, received)
+    return 1
+
```

Comment:
can we add a quick smoke test for the `ClassifySingle` and `ListDeployments` endpoints too? 

Can we make this run with pytest also? (should consume env vars and be marked as only running if the functional_test marker is enabled) that way we can put it into a pipeline _eventually_ when we're productionised properly. Ideally then we require it to merge but need to review the conditions around that & secrets etc.

Reply:
there's some method you can put in the main file that runs the stuff in pytest too that might be useful for this purpose

https://docs.pytest.org/en/stable/how-to/usage.html#calling-pytest-from-python-code